### PR TITLE
Pull project templates from repos

### PR DESCRIPTION
**Pull request by TCourtneyOwen**
Thank you for your pull request. Please provide the following information.

---

1. **Do these changes impact *User Experience*?** (e.g., how the user interacts with Yo Office and/or the files and folders the user sees in the project that Yo Office creates)
    > 
    > * [ ]  Yes
    > * [ ]  No

    If Yes, briefly describe what is impacted.

2. **Do these changes impact *documentation*?** (e.g., a tutorial on https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins)
    > 
    > * [X ]  Yes
    > * [ ]  No

    If Yes, briefly describe what is impacted.
Documentation will need to be updated to reflect the fact that users will no longer be prompted for the host type and that the project templates with the exception of manifest-only now live in their own repos

3. **Validation/testing performed**:
Ran automated tests locally to ensure they pass

    Describe manual testing done. 
Created all project types, both TS and JS, to ensure they install, build and work

4. **Platforms tested**:

    > * [X ] Windows
    > * [ ] Mac

### CODE REVIEW

**Comment by kbrandl** [comment]
@TCourtneyOwen thanks for indicating that the changes in this PR impact documentation. Are you already coordinating with someone on the docs team regarding alignment of timing for releasing these Yo Office changes and updating docs?

**Comment by TCourtneyOwen** [comment]
@kbrandl A group of us, including Reeza are meeting this week to discuss upcoming changes.  A more detailed description is available at https://microsoft.sharepoint.com/teams/MUIPlatform/Shared Documents/DevX (Developer Experience)/Debugging and Tooling/Yo Office Changes - Q1.dotx?web=1

**Comment by kbrandl** [comment]
@TCourtneyOwen excellent, thanks for confirming that. In that case, I'll trust that Reeza will make sure someone on his team updates the docs as needed.

File: package.json [hunk 1]
```diff
@@ -58,6 +58,7 @@
     "concurrently": "^3.4.0",
     "cpx": "^1.5.0",
     "mocha": "^5.2.0",
+    "office-addin-manifest": "^1.2.4",
```
Comment (akrantz) [inline package.json]: This should not be in both dependencies and devDependencies.

File: src/app/config/projectProperties.json [hunk 2]
```diff
@@ -1,70 +1,49 @@
 {
     "projectTypes": {
-        "angular": {
-            "displayname": "Office Add-in project using Angular framework",
+        "basic-taskpane": {
```
Comment (akrantz) [inline src/app/config/projectProperties.json]: suggest: "taskpane"

File: src/app/config/projectProperties.json [hunk 3]
```diff
@@ -1,70 +1,49 @@
...
+                    "repository": "https://github.com/OfficeDev/Office-Addin-TaskPane-JS"},
                 "typescript": {
-                    "repository": ""}
-                }     
+                    "repository": "https://github.com/OfficeDev/Office-Addin-TaskPane"}
+                }
         },
-        "excel-functions": {
-            "displayname": "Excel Custom Functions Add-in project (Requires the Insider channel for Excel)",
+        "angular": {
```
Comment (akrantz) [inline src/app/config/projectProperties.json]: Suggest: "angular-taskpane"

File: src/app/config/projectProperties.json [hunk 4]
```diff
@@ -1,70 +1,49 @@
 {
     "projectTypes": {
-        "angular": {
-            "displayname": "Office Add-in project using Angular framework",
+        "basic-taskpane": {
+            "displayname": "Office Add-in Taskpane project",
```
Comment (akrantz) [inline src/app/config/projectProperties.json]: we should get content team input about the display names...

Comment (akrantz) [reply src/app/config/projectProperties.json]: In the meantime, please make Taskpane two words in the display names, e.g. "Task Pane". This is how it appears in the docs.

File: src/app/index.ts [hunk 5]
```diff
@@ -266,10 +232,9 @@ module.exports = yo.extend({
     return new Promise((resolve, reject) => {
       try {
         let language = this.project.scriptType === typescript ? 'ts' : 'js';
-        const starterCode = generateStarterCode(this.project.host);
-        const templateFills = Object.assign({}, this.project, starterCode);
         let jsonData = new projectsJsonData(this.templatePath());
         let projectRepoBranchInfo = jsonData.getProjectRepoAndBranch(this.project.projectType, language);
+        const templateFills = Object.assign({}, this.project);
```
Comment (akrantz) [inline src/app/index.ts]: Is this still needed (for manifest only)?

File: src/app/index.ts [hunk 6]
```diff
@@ -358,7 +303,7 @@ module.exports = yo.extend({
     else
       {
         this.log('\n----------------------------------------------------------------------------------\n');
-        this.log(`      Creating ${chalk.bold.green(this.project.projectDisplayName)} add-in for ${chalk.bold.magenta(_.capitalize(this.project.host))} using ${chalk.bold.yellow(this.project.scriptType)} and ${chalk.bold.green(_.capitalize(this.project.projectType))} at ${chalk.bold.magenta(this._destinationRoot)}\n`);
+        this.log(`      Creating Office Taskpane Addin using ${chalk.bold.yellow(this.project.scriptType)} framework and ${chalk.bold.green(_.capitalize(this.project.projectType))} at ${chalk.bold.magenta(this._destinationRoot)}\n`);
```
Comment (akrantz) [inline src/app/index.ts]: Taskpane => Task Pane

File: package.json (hunk 1 above)
Comment (TCourtneyOwen) [reply package.json]: fixed

File: src/app/config/projectProperties.json (hunk 2 above)
Comment (TCourtneyOwen) [reply src/app/config/projectProperties.json]: fixed

File: src/app/index.ts (hunk 5 above)
Comment (TCourtneyOwen) [reply src/app/index.ts]: Yes, we are using this to set the manifest guid. Do you think we shoud just use modifyManifestFile for this? The reason I didn't use modifyManifestFile is that this restores the manifest to an imperfect state with comments removed

Comment (akrantz) [reply src/app/index.ts]: It's fine. Ideally we would fix modifyManifestFile to preserve comments, but this is fine for now.

File: src/app/index.ts (hunk 6 above)
Comment (TCourtneyOwen) [reply src/app/index.ts]: fixed

**Comment by LouMM** [comment]
This is a very large change list. As long as the team is fine with the individual PR/Commits, i think this merge is fine. We should have caught some of the typos and nit-picks in the smaller PR's. Something to look out for in the future. The PR looks fine, but I think a code refactor of this code base is coming in the near future. 👍

**Comment by TCourtneyOwen** [comment]
Most of the file changes are just deleted files (i.e. the various templates we are removing from generator-office).  Probably could have handled the actual required changes to pull from the repos in one PR and had another to track all the template files we are deleting.

**Comment by akrantz** [comment]
@TCourtneyOwen The final steps will need to be updated. Start the local web server should use "npm run dev-server" instead of "npm start".

**Comment by TCourtneyOwen** [comment]
@akrantz Regarding the comment "The final steps will need to be updated. Start the local web server should use "npm run dev-server" instead of "npm start".

Do we in fact want to add webpack and then add a "dev-server" script? I would prefer to steer clear of webpack if we can

**Comment by akrantz** [comment]
We don't need to add webpack. It is already using webpack-dev-server under the covers. But we need package.json script to have separate support for running the dev-server from start debugging. Hence:
{
    "dev-server": "react-scripts start", 
    "start": "office-addin-debugging start"
 }

**Comment by TCourtneyOwen** [comment]
I am going to close this PR and open a separate one without all the files removed to reduce confusion

**Pull request closed without merging**